Lib/test/test_smtplib.py

```diff
@@ -549,6 +549,34 @@ def testSendMessageMultipleResentRaises(self):
             smtp.send_message(m)
         smtp.close()
 
+    def testSendMessageAddDateIfMissing(self):
+        m = email.mime.text.MIMEText('A test message')
+        m['From'] = 'foo@bar.com'
+        m['To'] = 'John'
+        m['CC'] = 'Sally, Fred'
+        m['Bcc'] = 'John Root <root@localhost>, "Dinsdale" <warped@silly.walks.com>'
+        smtp = smtplib.SMTP(HOST, self.port, local_hostname='localhost', timeout=3)
+        smtp.send_message(m)
```

Comment:
Maybe we can mock up the `email.utils.formatdate()` to ensure we have the same result to tests.